builder/googlecompute/builder_acc_test.go

```diff
@@ -55,6 +61,76 @@ func TestAccBuilder_DefaultTokenSource(t *testing.T) {
 	acctest.TestPlugin(t, testCase)
 }
 
+// generateSSHPrivateKey generates a PEM encoded ssh private key file
+//
+// The file's deletion is the responsibility of the caller.
+func generateSSHPrivateKey() (string, error) {
```

Comment:
Nice test

I think we should also update this unit test so that public key is not set to nil. It should still pass  https://github.com/hashicorp/packer-plugin-googlecompute/blob/957770ecb4ca9f3508c0688d5421b30e7fa2f66a/builder/googlecompute/step_import_os_login_ssh_key_test.go#L200

Reply:
Good call! I've updated it with a generated private key file, and it indeed still passes